### ISSUE TITLE
Deploy deploy-cloud-functions ignores service account

### ISSUE THREAD

**Issue by ngreven-allegro** [google-github-actions/deploy-cloud-functions#448]
### TL;DR

When trying to use the deploy step in `deploy-cloud-functions`, it ignores the `service_account_email` setting.

We've created a new service account (SA) for our project in GCP. We've been trying to deploy it, but no matter what we do it will not deploy using the SA specified. It will always revert to the default SA (called 'Compute Engine default service account' on Google Cloud). The `service_account_email` seems to be ignored.

### Expected behavior

Setting the `service_account_email` in `deploy-cloud-functions` should use the specified SA instead of the standard one. 

I've tried using v3 of `deploy-cloud-functions` as well, replacing `service_account_email` with `service_account` instead. Unfortunately it gives me the exact same behaviour as described. 

Both `workload_identity_provider` and `service_account` in the 'auth' step are defined, but I changed them in this bug report for security reasons.

### Observed behavior

The default service account is used, `service_account_email` is ignored.

### Action YAML

```yaml
name: CI

on:
  push:
    branches: [ "master" ]

jobs:
  deploy:
    name: Deploy to GCP
    runs-on: [ self-hosted, ubuntu-latest ]
    container:
      image: artifactory-url/ubuntu:20.04
    permissions:
      contents: 'read'
      id-token: 'write'

    steps:
      - uses: actions/checkout@v4
        with:
          fetch-depth: 0
          ref: ${{ github.head_ref }}

      - id: 'auth'
        uses: 'google-github-actions/auth@v2'
        with:
          workload_identity_provider: 'xxx'
          service_account: 'yyy'

      - id: 'deploy'
        uses: 'google-github-actions/deploy-cloud-functions@v2'
        with:
          name: 'qqq'
          runtime: go122
          project_id: zzz
          region: europe-central2
          service_account_email: 'fff'
```

**Comment by sethvargo** [comment]
Hi there - could you please provide the [debug output](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging) for the complete GitHub Actions workflow run for v3?

**Comment by mrz-allegro** [comment]
https://github.com/google-github-actions/deploy-cloudrun?tab=readme-ov-file#authorization

Here I saw info for **deploy-cloudrun**, that the SA used to deploy needs to be a member of the Compute Engine default service account, with role Service Account User - which is exactly my case. Maybe it's the same for **deploy-cloud-functions**? (in that case some info in README would be appreciated)

**Comment by sethvargo** [comment]
> Here I saw info for deploy-cloudrun, that the SA used to deploy needs to be a member of the Compute Engine default service account, with role Service Account User - which is exactly my case. Maybe it's the same for deploy-cloud-functions? (in that case some info in README would be appreciated)

This is generally how GCP works. The identity that is being used to _deploy_ a workload must have the same permissions as the identity used to _run_ that workload.

**Comment by mrz-allegro** [comment]
Yes, but I don't use the Compute SA for any of those. First I authenticate as SA_1 to deploy the cloud function, and then I tell it to run as SA_2 by sending it in the 'service_account' parameter of the deploy-cloud-functions action. Both SA_1 and SA_2 are my custom Service Accounts, I don't want to use the default Compute SA at all.

**Comment by mrz-allegro** [comment]
I masked it with ".....".

In recent weeks I deployed a couple of different CFs using deploy-cloud-functions@v3 (GEN_2), in each case we used a different, dedicated runtime SA so the function would not use the default compute SA (as we don't want to use it at all - a company decision), and in each case I had to add the actAs permission for my deployment SA to both runtime SA and default compute SA - it wouldn't deploy otherwise.

**Comment by mrz-allegro** [comment]
Ok, I finally figured it out - turns out that I also have to set **build_service_account** and **event_trigger_service_account**, the default compute account was used there.

Sorry for bothering you.

**Comment by ngreven-allegro** [comment]
Completely forgot about opening this issue. I've spoken with my colleague, and it's been resolved. Closing this issue now.
